lib/Doctrine/ODM/MongoDB/Mapping/Annotations/File/Filename.php

```diff
@@ -4,19 +4,19 @@
 
 namespace Doctrine\ODM\MongoDB\Mapping\Annotations\File;
 
+use Attribute;
+use Doctrine\Common\Annotations\Annotation\NamedArgumentConstructor;
 use Doctrine\ODM\MongoDB\Mapping\Annotations\AbstractField;
 
 /**
  * @Annotation
+ * @NamedArgumentConstructor
  */
+#[Attribute(Attribute::TARGET_PROPERTY)]
 final class Filename extends AbstractField
 {
-    /** @var string */
-    public $name = 'filename';
-
-    /** @var string */
-    public $type = 'string';
-
-    /** @var bool */
-    public $notSaved = true;
+    public function __construct(?string $name = 'filename')
+    {
+        parent::__construct($name, 'string', false, [], null, true);
+    }
```

Comment:
```suggestion
    public function __construct()
    {
        parent::__construct('filename', 'string', false, [], null, true);
    }
```

The same reason than https://github.com/doctrine/mongodb-odm/pull/2344#discussion_r683185514